### PR TITLE
9/20

### PR DESCRIPTION
**Pull request by zerolfx**

### CODE REVIEW

File: src/chemotaxis/g8/Controller.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	}
+
+
+
+	public Controller(Point start, Point target, Integer size, ChemicalCell[][] grid, Integer simTime, Integer budget, Integer seed, SimPrinter simPrinter) {
+		super(start, target, size, grid, simTime, budget, seed, simPrinter);
+		shortestPath = generateShortestPath(start, target, size, grid);
+		selectedCells = selectCells(shortestPath, INTERVAL);
+
+		shift = 1;
```
Comment (samrmayers) [inline src/chemotaxis/g8/Controller.java]: Can we go over your notation a little? What does "shift" do?

File: src/chemotaxis/g8/Controller.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	 * @param budget      chemical budget
+	 * @param seed        random seed
+	 * @param simPrinter  simulation printer
+	 *
+	 */
+	static int[][] DIR = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
+	ArrayList<Point> shortestPath;
+	ArrayList<Point> selectedCells;
+	int shift;
+	static int INTERVAL = 4;
```
Comment (samrmayers) [inline src/chemotaxis/g8/Controller.java]: I think we should make "INTERVAL" a variable depending on what part of the map we are on (so we can apply more chemicals more frequently on a difficult part of the map) -- but we can do this in the future

**Review by samrmayers — APPROVED** [review]

File: src/chemotaxis/g8/Controller.java (hunk 1 above)
Comment (zerolfx) [reply src/chemotaxis/g8/Controller.java]: I renamed it to offset, and added some comments for that.

File: src/chemotaxis/g8/Controller.java (hunk 2 above)
Comment (zerolfx) [reply src/chemotaxis/g8/Controller.java]: I agree with you, and we can make our strategy more dynamic in the next version.

**Pull request merged**